package/MDAnalysis/core/universe.py

```diff
@@ -316,7 +313,8 @@ class Universe(object):
     """
     def __init__(self, topology=None, *coordinates, all_coordinates=False,
                  format=None, topology_format=None, transformations=None,
-                 guess_bonds=False, vdwradii=None, in_memory=False,
+                 guess_bonds=False, vdwradii=None, context='default',
+                 to_guess=[], in_memory=False,
```

Comment:
Never use a container as a default. If you change the container elsewhere, the default will change for the class.

Reply:
Better is to use None and the check for it

Reply:
Some more explanations here. The issue is with having a mutable object as the default value; this is due to how python handles variables and how it passes them to functions.

In short, the default value is created only once, and then all calls to the function will refer to the same object in memory if the argument is not provided. If you ever modify the object, it is modified for all the following calls of the function. Here is a caricatural example:

```python
def my_function(argument=[]):
    argument.append(42)
    return len(argument)


a = my_function()
print(f"a is {a}")
b = my_function()
print(f"b is {b}")
```
You can follow what happens in the code [on python tutor](https://pythontutor.com/visualize.html#code=def%20my_function%28argument%3D%5B%5D%29%3A%0A%20%20%20%20argument.append%2842%29%0A%20%20%20%20return%20len%28argument%29%0A%0A%0Aa%20%3D%20my_function%28%29%0Aprint%28f%22a%20is%20%7Ba%7D%22%29%0Ab%20%3D%20my_function%28%29%0Aprint%28f%22b%20is%20%7Bb%7D%22%29&cumulative=false&heapPrimitives=nevernest&mode=edit&origin=opt-frontend.js&py=3&rawInputLstJSON=%5B%5D&textReferences=false)

The code above outputs:
```
a is 1
b is 2
```

You can see that the result of the second call is affected by the first call. Often the issue is more subtle, but it makes for bugs that are difficult to figure out.

A general solution to the problem is the one suggested by @orbeckst: use `None` as the default and address the special case:

```python
def my_function(argument=None):
    if argument is None:
        argument = []  # we create a new list that is independent for every call
    ...
```

This solution has the benefit of being usable everywhere. However, it is not very expressive: when you see `None` as the default in the function signature you do not know how it will be treated. It also means you have to deal with special cases, which means more code and more potential errors.

In your specific case, you can avoid dealing with a special case. As a default value, you only need an empty iterator. If that empty iterator is not mutable you do not have the issue described above. An empty tuple is a non mutable iterator.

Here is the example above with a tuple (the code is a little different because tuples and lists do not have the same methods, however it would work without change in your case):

```python
def my_function(argument=()):
    argument = argument + (42,)
    return len(argument)


a = my_function()
print(f"a is {a}")
b = my_function()
print(f"b is {b}")
```

The code on [python tutor](https://pythontutor.com/visualize.html#code=def%20my_function%28argument%3D%28%29%29%3A%0A%20%20%20%20argument%20%3D%20argument%20%2B%20%2842,%29%0A%20%20%20%20return%20len%28argument%29%0A%0A%0Aa%20%3D%20my_function%28%29%0Aprint%28f%22a%20is%20%7Ba%7D%22%29%0Ab%20%3D%20my_function%28%29%0Aprint%28f%22b%20is%20%7Bb%7D%22%29&cumulative=false&heapPrimitives=nevernest&mode=edit&origin=opt-frontend.js&py=3&rawInputLstJSON=%5B%5D&textReferences=false).
The output:

```
a is 1
b is 1
```

Because you only iterate over the `to_guess` argument, and because iterating is the same on a list and on a tuple, you can use an empty tuple as the default value.